public/scripts/extensions/tts/cosyvoice.js

```diff
@@ -0,0 +1,204 @@
+import { getPreviewString, saveTtsProviderSettings } from './index.js';
+
+export { CosyVoiceProvider };
+
+class CosyVoiceProvider {
+    //########//
+    // Config //
+    //########//
+
+    settings;
+    ready = false;
+    voices = [];
+    separator = '. ';
+    audioElement = document.createElement('audio');
+
+    /**
+     * Perform any text processing before passing to TTS engine.
+     * @param {string} text Input text
+     * @returns {string} Processed text
+     */
+    processText(text) {
+        return text;
+    }
+
+    audioFormats = ['wav', 'ogg', 'silk', 'mp3', 'flac'];
+
+    languageLabels = {
+        'Auto': 'auto',
+    };
+
+    langKey2LangCode = {
+        'zh': 'zh-CN',
+        'en': 'en-US',
+        'ja': 'ja-JP',
+        'ko': 'ko-KR',
+    };
+
+    modelTypes = {
+        CosyVoice: 'CosyVoice',
+    };
+
+    defaultSettings = {
+        provider_endpoint: 'http://localhost:9880',
+        format: 'wav',
+        lang: 'auto',
+        streaming: false,
+
+    };
+
+    get settingsHtml() {
+        let html = `
+
+        <label for="tts_endpoint">Provider Endpoint:</label>
+        <input id="tts_endpoint" type="text" class="text_pole" maxlength="250" height="300" value="${this.defaultSettings.provider_endpoint}"/>
+        <br/>
```

Comment:
> It might be a good idea to include links to your forks in the UI so people would know exactly what to install. I guess the API endpoints are your own addition, and not from the original repo (FunAudioLLM)?

sure thing,Just added the project url,tks a lot